### DISCUSSION TITLE
RollerCoaster Tycoon Classic macOS Support

### DISCUSSION THREAD

**Discussion by dotvhs** [OpenRCT2/OpenRCT2#21767] [Feature requests] [answered]
I own RollerCoaster Tycoon Classic on Steam which has (broken) macOS native app, however OpenRCT2 doesn't seem to be compatible with it.

After opening OpenRCT2, I am greeted with this dialogue:
![CleanShot 2024-04-11 at 19 09 58@2x](https://github.com/OpenRCT2/OpenRCT2/assets/681416/793da377-3859-447a-acbd-cbaa9fa4562a)

After selecting my path to RollerCoaster Tycoon Classic on macOS, which is `/Users/<user>/Library/Application Support/Steam/steamapps/common/RollerCoaster Tycoon Classic`, I am presented with an error:
![CleanShot 2024-04-11 at 19 12 36@2x](https://github.com/OpenRCT2/OpenRCT2/assets/681416/37d811f2-2642-42d0-9398-c82a01f9d836)

Which is understandable because that file is not inside that folder directly, on macOS this folder looks like this:
![CleanShot 2024-04-11 at 19 14 30@2x](https://github.com/OpenRCT2/OpenRCT2/assets/681416/b5af4b05-0fbf-4b57-9194-0ac03e95e2e0)

And if I open the `.app` package, I can clearly see the `g1.dat` file but in a different path:
![CleanShot 2024-04-11 at 19 16 15@2x](https://github.com/OpenRCT2/OpenRCT2/assets/681416/58990177-8724-4565-b4e4-9137b127e2f8)

The complete path to that file with RollerCoaster Tycoon Classic for macOS would be:
`/Users/<user>/Library/Application Support/Steam/steamapps/common/RollerCoaster Tycoon Classic/RCT Classic.app/Contents/Resources/g1.dat`

Would it be possible to make OpenRCT2 compatible with it in the future? Thank you in advance.

**Comment by Gymnasiast** [comment]
This should not be too hard to implement. There is already some code to resolve the path for the Windows version of RCT Classic.

**Reply by baweaver** [reply]
Don't suppose this was looked into? I just hit a similar issue.

**Reply by Gymnasiast** [reply]
Not yet, I don’t daily drive macOS and I don’t own RCTC on Steam.

**Reply by Gymnasiast** [reply]
How similar is your issue, by the way?

**Reply by Gymnasiast** [reply, marked as answer]
I have now created https://github.com/OpenRCT2/OpenRCT2/pull/24179 . Would love to hear if that fixes your problems @dotvhs @baweaver
